Frontend/app/src/main/java/io/happybugs/happybugs/ReportActivity.java

```diff
@@ -0,0 +1,143 @@
+package io.happybugs.happybugs;
+
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.TextView;
+
+public class ReportActivity extends AppCompatActivity implements View.OnClickListener{
+    ReportButtons buttons;
+    ReportEditTexts editTexts;
+    ReportCheckBoxes checkBoxes;
+    ReportViews views;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_report);
+
+        buttons = new ReportButtons((Button) findViewById(R.id.whatBtn),
+                (Button) findViewById(R.id.whereBtn), (Button) findViewById(R.id.whenBtn),
+                (Button) findViewById(R.id.whoBtn), (Button) findViewById(R.id.detailsBtn),
+                (Button) findViewById(R.id.saveBtn));
+
+        editTexts = new ReportEditTexts((EditText) findViewById(R.id.whatText),
+                (EditText) findViewById(R.id.whereText), (EditText) findViewById(R.id.whenText),
+                (EditText) findViewById(R.id.whoText), (EditText) findViewById(R.id.detailsText));
+
+        checkBoxes = new ReportCheckBoxes((CheckBox) findViewById(R.id.whatCheck),
+                (CheckBox) findViewById(R.id.whereCheck), (CheckBox) findViewById(R.id.whenCheck),
+                (CheckBox) findViewById(R.id.whoCheck), (CheckBox) findViewById(R.id.detailsCheck));
+
+        views = new ReportViews((View) findViewById(R.id.whatView), (View) findViewById(R.id.whereView),
+                (View) findViewById(R.id.whenView), (View) findViewById(R.id.whoView),
+                (View) findViewById(R.id.detailsView));
+
+        buttons.whatBtn.setOnClickListener(this);
+        buttons.whereBtn.setOnClickListener(this);
+        buttons.whenBtn.setOnClickListener(this);
+        buttons.whoBtn.setOnClickListener(this);
+        buttons.detailsBtn.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()){
+            case R.id.whatBtn:
+                enableTextView(editTexts.whatText, editTexts.whereText,
+                        editTexts.whenText, editTexts.whoText, editTexts.detailsText);
+                enableBoxes(editTexts.whatText, checkBoxes.whatCheck, buttons.saveBtn);
+                enableView(editTexts.whatText, views.whatView,
+                        views.whereView, views.whenView, views.whoView, views.detailsView);
+                break;
```

Comment:
I see this block being repeated in all cases. can we create a function and refactor

Reply:
If I create a function for the repeated blocks it needs 21 parameters. Though not effective, I thought having repeated functions would be better for other readers since distinguishing 21 parameters is stressing (especially in this case where the variables are so confusing). Do you think it'd be better to refactor?

Reply:
Please see my comment on enableTextView function. Improving that function might naturally solve this issue.

Reply:
I think this is good enough now! Good job :)